src/components/views/admin/types.ts

```diff
@@ -0,0 +1 @@
+export type CandidateRateObj = { memberId: string; rating: number | null };
```

Comment:
When would the rating be `null` ??

Reply:
Initialized as null?

Reply:
This type represents the data that you send to the api. That data should **always** have a defined rating. 
The `newValue` you are getting from the `onChange` prop of the Rating component can be null. But if it is null I don't think you would send an appData. You should investigate which cases send you a null rating in the Rating component. To get by the typing of the `number | null` you just need to add a [type guard](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#typeof-type-guards) inside the function. Basically check for the truthiness of `newRating` with an if, inside the condition `newValue` will have type `number.